org.lflang/src/org/lflang/generator/ErrorReporter.kt

```diff
@@ -0,0 +1,150 @@
+            if (myResource === null && object !== null) {
+                // Attempt to identify the IResource from the object.
+                val eResource = object.eResource
+                if (eResource !== null) {
+                    val uri = FileConfig.toPath(eResource).toUri();
+                    myResource = getEclipseResource(uri);
+                }
+            }
+            // If the resource is still null, use the resource associated with
+            // the top-level file.
+            if (myResource === null) {
+                myResource = fileConfig.iResource
+            }
+            if (myResource !== null) {
+                val marker = myResource.createMarker(IMarker.PROBLEM)
+                marker.setAttribute(IMarker.MESSAGE, message);
+                if (line !== null) {
+                    marker.setAttribute(IMarker.LINE_NUMBER, line);
+                } else {
+                    marker.setAttribute(IMarker.LINE_NUMBER, 1);
+                }
+                // Human-readable line number information.
+                marker.setAttribute(IMarker.LOCATION, lineAsString);
+                // Mark as an error or warning.
+                marker.setAttribute(IMarker.SEVERITY, severity);
+                marker.setAttribute(IMarker.PRIORITY, IMarker.PRIORITY_HIGH);
+
+                marker.setAttribute(IMarker.USER_EDITABLE, false);
+
+                // NOTE: It might be useful to set a start and end.
+                // marker.setAttribute(IMarker.CHAR_START, 0);
+                // marker.setAttribute(IMarker.CHAR_END, 5);
+            }
+        }*/
+
+        // Return a string that can be inserted into the generated code.
+        return "[[$header: $message]]"
+    }
+
+    /**
+     * Report an error.
+     * @param message The error message.
+     */
+    override fun reportError(message: String): String = report(message, IMarker.SEVERITY_ERROR, null, null)
+
+    /**
+     * Report a warning.
+     * @param message The warning message.
+     */
+    override fun reportWarning(message: String): String = report(message, IMarker.SEVERITY_WARNING, null, null)
+
+    /**
+     * Report an error on the receiving parse tree object.
+     *
+     * @param obj     The parse tree object.
+     * @param message The error message.
+     */
+    override fun reportError(obj: EObject, message: String): String = report(message, IMarker.SEVERITY_ERROR, obj)
+
+    /**
+     * Report a warning on the receiving parse tree object.
+     *
+     * @param obj     The parse tree object.
+     * @param message The error message.
+     */
+    override fun reportWaring(obj: EObject, message: String): String = report(message, IMarker.SEVERITY_WARNING, obj)
+
+    /** Report an error on the receiving parse tree object.
+     *  @receiver The parse tree object.
+     *  @param message The error message.
+     */
+    fun EObject.extReportError(message: String): String = report(message, IMarker.SEVERITY_ERROR, obj = this)
```

Comment:
This is not relevant anymore because I removed the extension function from the interface. However, I had the problem that the java `IErrorReporter` interface (or `ErrorReporter`) now, defined the method `String reportError(EObject obj, String message)`. The Kotlin implementation of this interface overrides this method. Now Kotlin does not allow me to define an additional extension function with the same name. Hence, the `ext` prefix.

For now, I simply removed the extensions, because it doesn't seem possible to define them as part of the Java interface.